DependencyInjection/Configuration.php

```diff
@@ -41,7 +41,7 @@ public function getConfigTreeBuilder() : TreeBuilder
 
         $rootNode
             ->children()
-                ->scalarNode('dir_name')->defaultValue('%kernel.root_dir%/DoctrineMigrations')->cannotBeEmpty()->end()
+                ->scalarNode('dir_name')->defaultValue('%kernel.project_dir%/DoctrineMigrations')->cannotBeEmpty()->end()
```

Comment:
In many cases, this now changes the default value if an application is based on the Symfony Standard Edition.

Reply:
Isnt project dir and root dir the same? If not, than this should be considered a BC break.

Reply:
This old blog about a Symfony 3 and 4 version explains it more in detail: https://symfony.com/blog/new-in-symfony-3-3-a-simpler-way-to-get-the-project-root-directory

This seems like a BC break when DoctrineMigrationsBundle gets updated to a version with these changes.

Reply:
So, What should I do in this situation?

Reply:
@xabbuh what's the best way to deal with this then? Should we move this change to the next major version and document it as a BC break?

Reply:
What if we move the logic to php for deciding which parameter to use?

Like the following

```diff
diff --git a/DependencyInjection/Configuration.php b/DependencyInjection/Configuration.php
index c7adbdb..b5d3ff1 100644
--- a/DependencyInjection/Configuration.php
+++ b/DependencyInjection/Configuration.php
@@ -41,7 +41,7 @@ class Configuration implements ConfigurationInterface
 
         $rootNode
             ->children()
-                ->scalarNode('dir_name')->defaultValue('%kernel.root_dir%/DoctrineMigrations')->cannotBeEmpty()->end()
+                ->scalarNode('dir_name')->defaultValue('%doctrine.migrations.dir%')->cannotBeEmpty()->end()
                 ->scalarNode('namespace')->defaultValue('Application\Migrations')->cannotBeEmpty()->end()
                 ->scalarNode('table_name')->defaultValue('migration_versions')->cannotBeEmpty()->end()
                 ->scalarNode('column_name')->defaultValue('version')->end()
diff --git a/DependencyInjection/DoctrineMigrationsExtension.php b/DependencyInjection/DoctrineMigrationsExtension.php
index 4a0e923..640c80b 100644
--- a/DependencyInjection/DoctrineMigrationsExtension.php
+++ b/DependencyInjection/DoctrineMigrationsExtension.php
@@ -23,6 +23,12 @@ class DoctrineMigrationsExtension extends Extension
     {
         $configuration = new Configuration();
 
+        if ($container->getParameterBag()->has('kernel.root_dir')) {
+            $container->setParameter('doctrine.migrations.dir', '%kernel.root_dir%/DoctrineMigrations');
+        } else {
+            $container->setParameter('doctrine.migrations.dir', '%kernel.project_dir%/src/Migrations');
+        }
+
         $config = $this->processConfiguration($configuration, $configs);
 
         foreach ($config as $key => $value) {
```

This would only change the default for Symfony 5+ apps where the parameter `%kernel.root_dir%` does not exist. Or am I missing something? :blush:

Reply:
@dmaicher that produces the same default across all Symfony version, thus preserves BC :+1: 

related to https://github.com/symfony/recipes/issues/718 it could be considerable to deprecate _not_ setting `dir_name` (which should effect a small group, as the recipe already sets it since forever in most projects AFAIK).

That would allow doctrine to update the default in its next major; the recipe is proposing `%kernel.project_dir%/migrations`. So if we can keep the default in sync, we can upgrade the recipe (remove the config) once we depend on doctrine's next major.

Reply:
We're making more significant changes in the next major, where there may be multiple paths for migrations, so I'm not too concerned about the upcoming BC break.

@dmaicher great idea with the change, that should work since we can check what version we're on and select the appropriate parameter. @michaljusiega can you make those changes please? The patch above should suffice to ensure BC.

Reply:
@alcaeus done, waiting for green CI